src/dashboard/apps/consent/tests/test_views.py

```diff
@@ -0,0 +1,156 @@
+"""Dashboard consent views tests."""
+
+import pytest
+from django.urls import reverse
+
+from apps.auth.factories import UserFactory
+from apps.consent import AWAITING, VALIDATED
+from apps.consent.factories import ConsentFactory
+from apps.consent.models import Consent
+from apps.consent.views import ConsentFormView
+from apps.core.factories import DeliveryPointFactory, EntityFactory
+from apps.core.models import DeliveryPoint
+
+
+@pytest.mark.django_db
+def test_bulk_update_consent_status_without_ids(rf):
+    """Test that no status is updated if no id is passed."""
+    user1 = UserFactory()
+
+    request = rf.get(reverse("consent:manage"))
+    request.user = user1
+
+    view = ConsentFormView()
+    view.setup(request)
+
+    entity1 = EntityFactory(users=(user1,), name="entity1")
+    for i in range(1, 4):
+        DeliveryPointFactory(provider_assigned_id=f"entity1_{i}", entity=entity1)
+
+    # create awaiting consent for each delivery points
+    ids = []
+    for delivery_point in DeliveryPoint.objects.all():
+        consent = ConsentFactory(delivery_point=delivery_point, created_by=user1)
+        ids.append(str(consent.id))
```

Comment:
In the `ConsentFactory` you can also define a [`RelatedFactory`](https://factoryboy.readthedocs.io/en/stable/reference.html#relatedfactory) (for the `DeliveryPoint`) to simply consistent datasets generation.

Reply:
Alternatively use the existing `SubFactory`?